### PR TITLE
Style improvements after review

### PR DESCRIPTION
**Pull request by hello1101n**

### CODE REVIEW

**Comment by hello1101n** [comment]
Each llist item starts with line number(s).
1. 8: plat? Why not use an existing word?)
2. 10,11: Lines can be merged in one import.
3. 29: Why do you comment this code in comments?
4. 41: Can be replaced with Promise.all(...).catch(...).
5. 43: const process = files.map(...).filter(...).
6. 49: try...catch inside of then can be replaced with then(...).catch(...).
7. 92: Sometimes C:/Windows/ isn't a path to the Windows directory. Using $PATH can solve this problem.
8. 93: Hacky trim, split, slice, and crowded map can be replaced with split, chain of maps, and filter.
9. 113: Wrapping switch in the try...catch feels grumpy. I will show an alternative later.
TBH on the first sign this file feels like a bunch of copy-pasted code fragments with chaotic structure and tricky data-flow.
I think each function (both of them, yes) in this file should only choose platform specific methods from separate files and handle some errors in unified way.
Smth like this:
```
export async function killProcByPID(pidString) {
  const pid = parseInt(pidSting, 10);

  if (!Number.isInteger(pid)) {
    return {
      result: null, // []?
      error: 'PID is not a number' };
  }
  
  const killByPIDFn;

  switch (platform) {
    case 'win32':
      killByPIDFn = require('./platform/win32').killByPID;       
    case 'linux':
    case 'darwin':
      killByPIDFn = require('./platform/unix').killByPID;
    default:
      killByPIDFn = () => throw new Error(OSErrMsg);
  }

  return await killByPIDFn(pid)
    .then(({stdout, stderr}) => {
      result: stdout,
      error: stderr
    })
    .catch(e => {
      result: null, // []?
      error: e.stderr || e.message
    })
}
```
Switch...case in this example can be replaced with a decision map ofc.
This is an draft of directory structure:
```
└── package-dir
    ├── index.js
    └── platform
        ├── darwin.js
        ├── win32.js
        └── unix.js
```
I like your attempt of extracting getLinuxProc. All other functions should be extracted from switch...case too. (edited)

**Comment by evgeny-w** [comment]
Regarding (3), that is the jsdoc syntax to declare an example usage of a function.
I am confused what you mean by (6), I am not catching error in the promise, I am catching error from lines 50-53, or did I implement this wrongly?

(7),(8) I will fix these but we are moving to a non-shell based solution, so this will become obsolete. I was working on the npm-package-c branch, please take a look.

Regarding (9), sorry, I am inexperienced in nodejs (this is my first time working with it) so I do not know the proper syntax/workflow for it - thanks for showing me the standard way to do it. However, all the code written here is mine, or re-adapted from another language to javascript by me (specifically, getLinuxProc from go-ps). The reason I wrapped try catch around the switch is for ease of reading - it is more visible where the error comes from instead of having to handle it inside of every case. 

Are you sure that splitting functions into different files is the best solution? Those files will have very few lines of code and obscure the code behind them inside of the main js file. Anyway, this is only the protoype solution, after moving to the native module solution functions will be imported from their own relevant files, this is already the case on npm-package-c branch (for windows at least, I am working on other platforms). I thought that visibility would be the most important for this case.

Also, the solution of creating a variable to hold a function that can change every time and then return it doesn't feel right to me, is this the standard for nodejs? It looks like it will also be "required" every time this function is called, does this cause any memory issues? I spent some time trying to change what modules are require() based on the platform of the user, but this turned out to be difficult, with the only solution being creating a variable and assigning require() to it, what also made my linter complain, so I thought this was not standard. I think your solution does look nicer though, perhaps if the files were required at the top of the file and not dynamically inside of the function. 

Speaking of linters, could you please reccomend me the format that you are using? I have been following Airbnb style but it looks like you are following another standard, could you please tell me which one to use?

File: lib/index.js [hunk 1]
```diff
@@ -87,13 +86,17 @@ async function getLinuxProc() { // In progress. Not tested
...
       case 'win32':
-        const winResult = await execProm('C:/Windows/System32/tasklist.exe /FO CSV');
+        const winResult = await execProm('tasklist /FO CSV');
         const winProcesses = winResult.stdout.trim()
           .split('\r\n')
-          .slice(1)
-          .map((x) => x.slice(1).trim().split('","').slice(0, 2)
+          .slice(1)// Remove "Image Name","PID","Session Name","Session#","Mem Usage" element
+          .map((process) => process
+            .slice(1)
```
Comment (He4eT) [inline lib/index.js]: Удаяляем первый символ? оО

File: lib/index.js [hunk 2]
```diff
@@ -156,31 +156,29 @@ async function getProcList() {
...
-  if (!Number.isInteger(tempPid)) {
+  if (!Number.isInteger(pid)) {
     return { result: null, error: 'PID is not a number' };
   }
 
   try {
-    switch (plat) {
+    switch (platform) {
       case 'win32':
-        const winResult = await execProm(`C:/Windows/System32/taskkill /F /PID ${tempPid}`);
```
Comment (hello1101n) [inline lib/index.js:169]: taskkill

Comment (hello1101n) [reply lib/index.js:169]: check if not - ok

File: lib/index.js [hunk 3]
```diff
@@ -87,13 +86,17 @@ async function getLinuxProc() { // In progress. Not tested
...
-        const winResult = await execProm('C:/Windows/System32/tasklist.exe /FO CSV');
+        const winResult = await execProm('tasklist /FO CSV');
         const winProcesses = winResult.stdout.trim()
           .split('\r\n')
-          .slice(1)
-          .map((x) => x.slice(1).trim().split('","').slice(0, 2)
+          .slice(1) // Remove "Image Name","PID","Session Name","Session#","Mem Usage" element
+          .map((process) => process
+            .trim()
+            .split(',')
```
Comment (hello1101n) [inline lib/index.js]: change split and trim

**Pull request merged**